crates/rattler_cache/src/run_exports_cache/download.rs

```diff
@@ -0,0 +1,66 @@
+use std::sync::Arc;
+
+use ::tokio::io::{AsyncSeekExt, AsyncWriteExt};
+use fs_err::tokio;
+use futures::StreamExt;
+use rattler_package_streaming::DownloadReporter;
+use tempfile::NamedTempFile;
+use url::Url;
+
+/// Download the contents of the archive from the specified remote location
+/// and store it in a temporary file.
+pub(crate) async fn download(
+    client: reqwest_middleware::ClientWithMiddleware,
+    url: Url,
+    suffix: &str,
+    reporter: Option<Arc<dyn DownloadReporter>>,
+) -> Result<NamedTempFile, DownloadError> {
+    let temp_file = NamedTempFile::with_suffix(suffix)?;
+
+    // Send the request for the file
+    let response = client.get(url.clone()).send().await?.error_for_status()?;
+
+    if let Some(reporter) = &reporter {
+        reporter.on_download_start();
+    }
+
+    let total_bytes = response.content_length();
+    // Convert the named temp file into a tokio file
+    let mut file = tokio::File::from_std(fs_err::File::from_parts(
+        temp_file.as_file().try_clone()?,
```

Comment:
You can use NamedTempFile into_parts to get the file handle without cloning. Cloning may fail.

At the end of the function you can call from_parts to reconstruct the namedtempfile